### DISCUSSION TITLE
与Valine的兼容问题

### DISCUSSION THREAD

**Discussion by wznmickey** [walinejs/waline#66] [Q&A] [answered]
在可视化的界面中无法显示之前在valine中发出的评论。

**Comment by lizheming** [comment]
是说后台无法无法显示么？正常来说不会的，可以提供下操作和截图我看一下。

**Reply by wznmickey** [reply]
![image](https://user-images.githubusercontent.com/44784663/102156955-997e5f80-3eb9-11eb-8e45-8a3ddccb8de7.png)
只有这一条是在服务器端输入的。
在博客页面数据好像一直被拦截。
![image](https://user-images.githubusercontent.com/44784663/102157060-c5014a00-3eb9-11eb-918f-7b2707ba57c8.png)
谢谢。

**Reply by lizheming** [reply]
后面的那个错误你把 serverURL 最后的那个 `/` 去掉

**Reply by wznmickey** [reply]
非常感谢，博客的访问正常了

**Reply by lizheming** [reply]
客气，这个问题我之后再看一下吧，看能否做下兼容

**Comment by wznmickey** [comment, marked as answer]
谢谢




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Austin Lee"<notifications@github.com&gt;; 
发送时间: 2020年12月15日(星期二) 上午9:49
收件人: "lizheming/waline"<waline@noreply.github.com&gt;; 
抄送: "Mickey"<first@wznmickey.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [lizheming/waline] 与Valine的兼容问题 (#66)





 
客气，这个问题我之后再看一下吧，看能否做下兼容
 
&mdash;

**Reply by lizheming** [reply]
所以后台还是不显示老数据是吗？请问前台页面是否显示老数据？

**Reply by wznmickey** [reply]
前台有的，后台没有
/comment?type=list&owner=all&status=approved&keyword=&page=1 返回的是 {"errno":401,"errmsg":"Unauthorized"}

**Reply by lizheming** [reply]
接口返回 401 状态码表示的是用户没有登录，或者未找到该用户。而且如果是未登录的话应该进入不了后台，也不会说只显示一条评论，老的评论不展示的。

**Reply by wznmickey** [reply]
对不起，是我搞错了，返回有数据的，但是只有新的数据

**Reply by wznmickey** [reply]
发现问题了，在leancloud里，valine没有设置status这一项，waline的ui读取需要这一项为approved，所以之前valine的读不进去

**Reply by lizheming** [reply]
@wznmickey 哦，是的 status 这一项是后来为了区分是否是垃圾评论加入的，所以可能有问题，我这边兼容处理下吧，status 不存在或者为 approved 的都是通过的评论。https://github.com/lizheming/waline/issues/67 提了个 issue 可以关注下这块的状态。

**Reply by lizheming** [reply]
你好，`@waline/vercel@0.13.4` 版本已经修复了这个问题，你可以更新下版本看一下。Vercel 部署的话可以直接将仓库中的 `package.json` 中的 `@waline/vercel` 版本修改成 0.13.4 进行更新。

**Reply by wznmickey** [reply]
> 你好，`@waline/vercel@0.13.4` 版本已经修复了这个问题，你可以更新下版本看一下。Vercel 部署的话可以直接将仓库中的 `package.json` 中的 `@waline/vercel` 版本修改成 0.13.4 进行更新。

非常感谢，问题解决了
